File: automation.js

const { Builder, By, until} = require('selenium-webdriver');
const firefox = require('selenium-webdriver/firefox');
const nodemailer = require('nodemailer');

// Configure the email transporter
let transporter = nodemailer.createTransport({
    service: 'gmail',
    auth: {
        user: 'alijone754rats@gmail.com',
        pass: 'eqey mexg bmlp kjhl'
    }
});

// Function to send email
async function sendEmail(subject, text, recipientEmail) {
    let mailOptions = {
        from: 'alijone754rats@gmail.com',
        to: recipientEmail,
        subject: subject,
        text: text
    };

    transporter.sendMail(mailOptions, function (error, info) {
        if (error) {
            console.log('Error sending email:', error);
        } else {
            console.log('Email sent: ' + info.response);
        }
    });
}

async function automateLogin(username, password, recipientEmail) {
    let options = new firefox.Options();
    // For headless mode (no GUI), uncomment the next line
    options.headless();

    let driver = await new Builder()
        .forBrowser('firefox')
        .setFirefoxOptions(options)
        .build();

    try {
        await driver.get('https://merlinswood.flowhcm.com/#/signin');
        // [Rest of your selenium code]
        // await driver.sleep(10000);
        //wait until loading element is gone
        await driver.wait(async () => {
            const elements = await driver.findElements(By.className('c-spinner2'));
            if (elements.length === 0) {
                return true;
            }
            const style = await elements[0].getAttribute('style');
            return style.includes('display: none');
        }, 100000);

        let signOutButtonPresent = await driver.findElements(By.className('btn-SignOut')).then(elements => elements.length > 0);

        if (signOutButtonPresent) {
            await driver.findElement(By.className('profile-box')).click();
            let lastDropdownItem = await driver.wait(until.elementLocated(By.css('.dropdown-menu li.last')), 10000);
            await lastDropdownItem.click();
            await driver.wait(until.elementLocated(By.id('username')), 10000);
        }

        await driver.wait(until.elementLocated(By.id('username')), 100000);
        let usernameInput = await driver.findElement(By.id('username'));
        await usernameInput.sendKeys(username);
        await driver.findElement(By.id('password')).sendKeys(password);
        await driver.findElement(By.className('sign-in-btn1')).click();
        await driver.wait(until.elementLocated(By.className('btn-SignOut')), 100000);
        // await driver.wait(until.elementLocated(By.className('sign-btn')), 100000);
        await driver.wait(async () => {
            const elements = await driver.findElements(By.className('modal-backdrop'));
            if (elements.length === 0) {
                return true;
            }
            const style = await elements[0].getAttribute('style');
            return style.includes('display: none');
        }, 10000);
        await driver.findElement(By.className('btn-SignOut')).click();
        // await driver.findElement(By.className('sign-btn')).click();
        await driver.sleep(10000);
        await sendEmail("Login Successful", "Your login was successful for account: " + username, recipientEmail);

    } catch (error) {
        await sendEmail("Login Error", "An error occurred during login for account: " + username + ". Error: " + error.message, recipientEmail);
    } finally {
        await driver.quit();
    }
}

const credentials = [
    { username: 'ali.jone@deltabluecarbon.com', password: 'Ali123', email: 'a.jone.23031@khi.iba.edu.pk' },
    // { username: 'ali.haider@deltabluecarbon.com', password: 'Ali@123', email: 'm.haider.23047@khi.iba.edu.pk' },
    // { username: 'muhammad.arham@deltabluecarbon.com', password: 'Arham123', email: 'arrhamjohar.10@gmail.com' },
    // ... other credentials
];

async function runScriptForMultipleAccounts() {
    for (const credential of credentials) {
        await automateLogin(credential.username, credential.password, credential.email);
    }
}

runScriptForMultipleAccounts();

File: automation.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.chrome.service import Service
from selenium.webdriver.support import expected_conditions as EC
from webdriver_manager.chrome import ChromeDriverManager
from selenium.webdriver.chrome.options import Options
import smtplib
from email.mime.text import MIMEText
from email.mime.multipart import MIMEMultipart

from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.chrome.options import Options
from selenium.webdriver.support import expected_conditions as EC
import smtplib
from email.mime.text import MIMEText
from email.mime.multipart import MIMEMultipart

# Configure the email transporter
def send_email(subject, text, recipient_email, server):
    sender_email = "alijone754rats@gmail.com"
    sender_password = "eqey mexg bmlp kjhl"  # Consider a more secure way to store this

    message = MIMEMultipart()
    message["From"] = sender_email
    message["To"] = recipient_email
    message["Subject"] = subject

    message.attach(MIMEText(text, "plain"))

    try:
        server.sendmail(sender_email, recipient_email, message.as_string())
        print("Email sent successfully")
    except Exception as e:
        print(f"Error sending email: {e}")

# Function to automate login
def automate_login(username, password, recipient_email, server):
    chrome_options = Options()
    # Configure WebDriver
    service = Service(ChromeDriverManager().install())
    driver = webdriver.Chrome(service=service, options=chrome_options)
    driver.implicitly_wait(10)

    try:
        driver.get("https://merlinswood.flowhcm.com/#/signin")

        WebDriverWait(driver, 10).until(EC.presence_of_element_located((By.ID, "username")))
        driver.find_element(By.ID, "username").send_keys(username)
        driver.find_element(By.ID, "password").send_keys(password)
        driver.find_element(By.CLASS_NAME, "sign-in-btn1").click()

        WebDriverWait(driver, 10).until(EC.presence_of_element_located((By.CLASS_NAME, "btn-SignOut")))
        WebDriverWait(driver, 10).until(lambda d: len(d.find_elements(By.CLASS_NAME, "modal-backdrop")) == 0 or "display: none" in d.find_element(By.CLASS_NAME, "modal-backdrop").get_attribute("style"))
        driver.find_element(By.CLASS_NAME, "btn-SignOut").click()

        send_email("Login Successful", "Your login was successful for account: " + username, recipient_email, server)

    except Exception as e:
        send_email("Login Error", "An error occurred during login for account: " + username + ". Error: " + str(e), recipient_email, server)
    finally:
        driver.quit()

# Credentials list
credentials = [
    {"username": "ali.jone@deltabluecarbon.com", "password": "Ali123", "email": "a.jone.23031@khi.iba.edu.pk"},
    # ... other credentials
]

# Run script for multiple accounts
def run_script_for_multiple_accounts():
    server = smtplib.SMTP('smtp.gmail.com', 587)
    server.starttls()
    server.login("alijone754rats@gmail.com", "eqey mexg bmlp kjhl")  # Use the secure method for password

    for credential in credentials:
        automate_login(credential["username"], credential["password"], credential["email"], server)

    server.quit()

run_script_for_multiple_accounts()
